tests/src/Integration/StaffTest.php

```
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Integration;
+
+use App\Tests\Entity\StaffTest\Staff;
+use Doctrine\Persistence\ObjectRepository;
+
+class StaffTest extends IntegrationTestCase
+{
+    private ObjectRepository $staffRepository;
+
+    /** @psalm-suppress PossiblyNullReference, MixedMethodCall */
+    public function testEntityFromRepoReturnsCorrectValue(): void
+    {
+        /** @var Staff $dora */
+        $dora = $this->staffRepository->find('4');
+
+        $this->assertSame(
+            'Dora',
+            $dora->getName(),
+            "An unexpected value was retrieved from a 'staff' entity."
+        );
+    }
+
+    /** @psalm-suppress PossiblyNullReference, MixedMethodCall */
+    public function testAddedEntityReturnsCorrectValue(): void
+    {
+        $name = 'Muttley';
+        $occupation = 'sidekick';
+
+        $muttley = new Staff();
+        $muttley->setName($name);
+        $muttley->setOccupation($occupation);
+        $this->entityManager->persist($muttley); /** @phpstan-ignore-line */
+        $this->entityManager->flush(); /** @phpstan-ignore-line */
+
+        /** @var Staff $dbMuttley */
+        $dbMuttley = $this->staffRepository->findOneBy(
+            [
+                'Name' => $name,
+                'Occupation' => $occupation
+            ]
+        );
+
+        $this->assertSame(
+            $dbMuttley->getName(),
+            $muttley->getName(),
+            'we were unable to persist an enity and retrieve its values'
+        );
+    }
+
+    /** @psalm-suppress PossiblyNullArgument, MixedArgument, PossiblyNullReference  */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        #@todo we could throw an exception here if getRepository returns null
```

What do we want to do about this?

---

I'd say two options:
1/ throw an exception, report that the repository is null, test will fail
2/ if the repository is null, tests will fail anyway and an exception will probably have been thrown anyway, so perhaps simpler to just let the test fail and not complicate things by explicitly throwing an exception

---

opted for option 2....